# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1492467280 14400
#      Mon Apr 17 18:14:40 2017 -0400
# Node ID aab07d7e98be8c472ee0f0b511bb8a1b39b1ac86
# Parent  32ec90068be51515400aac13a6ff0019efd78ed4
make testif code and docs match

* testfun.txi, test.m: Document proper placement of <bug-id> in
   %!testif blocks.

diff --git a/doc/interpreter/testfun.txi b/doc/interpreter/testfun.txi
--- a/doc/interpreter/testfun.txi
+++ b/doc/interpreter/testfun.txi
@@ -333,20 +333,23 @@ PASSES 3 out of 4 tests (1 expected fail
 @table @code
 @item %!test
 @itemx %!test <MESSAGE>
 Check that entire block is correct.  If @code{<MESSAGE>} is present, the
 test block is interpreted as for @code{xtest}.
 
 @item %!testif HAVE_XXX
 @itemx %!testif HAVE_XXX, HAVE_YYY, @dots{}
+@itemx %!testif HAVE_XXX, HAVE_YYY @dots{}; RUNTIME_COND
 @itemx %!testif @dots{} <MESSAGE>
-Check block only if Octave was compiled with feature HAVE_XXX.  If
-@code{<MESSAGE>} is present, the test block is interpreted as for
-@code{xtest}.
+Check block only if Octave was compiled with feature @code{HAVE_XXX}.
+@code{RUNTIME_COND} is an optional expression to evaluate to check
+whether some condition is met when the test is executed.  If
+@code{RUNTIME_COND} is false, the test is skipped.  If @code{<MESSAGE>}
+is present, the test block is interpreted as for @code{xtest}.
 
 @item %!xtest
 @itemx %!xtest <MESSAGE>
 Check block, report a test failure but do not abort testing.  If
 @code{<MESSAGE>} is present, then the text of the message is displayed
 if the test fails, like this:
 
 @example
diff --git a/scripts/testfun/test.m b/scripts/testfun/test.m
--- a/scripts/testfun/test.m
+++ b/scripts/testfun/test.m
@@ -524,18 +524,18 @@ function [__n, __nmax, __nxfail, __nbug,
           end_try_catch
           clear __test__;
         endif
         ## Code already processed.
         __code = "";
 
 ### TESTIF HAVE_FEATURE
 ### TESTIF HAVE_FEATURE ; RUNTIME_CONDITION
-### TESTIF <bug-id> HAVE_FEATURE
-### TESTIF <bug-id> HAVE_FEATURE ; RUNTIME_CONDITION
+### TESTIF HAVE_FEATURE <bug-id>
+### TESTIF HAVE_FEATURE ; RUNTIME_CONDITION <bug-id>
 ###
 ###   HAVE_FEATURE is a comma- or whitespace separated list of
 ###   macro names that may be checked with __have_feature__.
 ###
 ###   RUNTIME_CONDITION is an expression to evaluate to check
 ###   whether some condition is met when the test is executed.  For
 ###   example, have_window_system.
 
